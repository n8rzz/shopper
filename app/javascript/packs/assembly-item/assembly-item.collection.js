export default class AssemblyItemCollection {
    get length() {
        return this._items.length;
    }

    constructor() {
        this._items = [];

        return this._init();
    }

    _init() {
        return this;
    }

    add(viewModel) {
        this._items.push(viewModel);
    }

    findById(itemId) {
        return this._items.filter((item) => item.itemId === itemId)[0];
    }
}
