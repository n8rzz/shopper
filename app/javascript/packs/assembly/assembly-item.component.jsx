import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const BUTTON_VALUE_MAP = {
    DEFAULT: 'Add to order',
    IS_SUBMITTING: 'Adding...',
    SUCCESS: 'Success!',
};

export class AssemblyItem extends React.Component {
    _buildAssemblyButtonText() {
        if (this.props.isSubmitAssemblySuccess) {
            return BUTTON_VALUE_MAP.SUCCESS;
        }

        return !this.props.isSubmittingAssembly
            ? BUTTON_VALUE_MAP.DEFAULT
            : BUTTON_VALUE_MAP.IS_SUBMITTING;
    }

    _buildAssemblyItemButtonText(assemblyItemId) {
        if (
            this.props.isAssemblyItemSubmitSuccess
            && this.props.submittedAssemblyItemId === assemblyItemId
        ) {
            return BUTTON_VALUE_MAP.SUCCESS;
        }

        return this.props.submittedAssemblyItemId !== assemblyItemId
            ? BUTTON_VALUE_MAP.DEFAULT
            : BUTTON_VALUE_MAP.IS_SUBMITTING;
    }

    _buildAssemblyButtonClassnames() {
        return classnames({
            btn: true,
            'btn-toggle': true,
            'mix-btn_isLoading': this.props.isSubmittingAssembly,
            'mix-btn_isSuccess': this.props.isSubmitAssemblySuccess,
        });
    }

    _buildAssemblyItemButtonClassnames(assemblyItemId) {
        const isSuccess = this.props.isAssemblyItemSubmitSuccess
            && this.props.submittedAssemblyItemId === assemblyItemId;

        return classnames({
            btn: true,
            'btn-toggle': true,
            'mix-btn_isLoading': !isSuccess && this.props.submittedAssemblyItemId === assemblyItemId,
            'mix-btn_isSuccess': isSuccess,
        });
    }

    _buildAssemblyGroupHeaderJsx() {
        const assemblyItemList = this.props.assemblyItemsGroupedByAssemblyId[this.props.assembly.id];
        const buttonText = this._buildAssemblyButtonText();

        return (
            <div className={`listItem js-assembly-${this.props.assembly.id}`}>
                <div className={'listItem-hd'}>
                    <h2
                        className={'hdg hdg_2'}
                        onClick={this.props.onClickAccordionTriggerHandler}
                    >
                        {this.props.assembly.name}
                    </h2>
                    <div>
                        <button
                            type={'button'}
                            className={this._buildAssemblyButtonClassnames()}
                            data-assembly-id={this.props.assembly.id}
                            onClick={this.props.onClickAddAssemblyHandler}
                        >
                            {buttonText}
                        </button>
                    </div>
                </div>
                <div className={'listItem-ft'}>
                    <ul className={'stereo'}>
                        <li>
                            Items:
                            {assemblyItemList.length}
                        </li>
                        <li>
                            <a
                                className={'link'}
                                href={`/assemblies/${this.props.assembly.id}/edit`}
                            >
                                Edit Assembly
                            </a>
                        </li>
                    </ul>
                </div>
            </div>
        );
    }

    _buildAssemblyItemList() {
        const assemblyItemList = this.props.assemblyItemsGroupedByAssemblyId[this.props.assembly.id];
        const assemblyItemListJsx = assemblyItemList.map((assemblyItem, index) => {
            const item = this.props.itemIdMap[assemblyItem.item_id];
            const departmentName = this.props.departmentIdMap[item.department_id].name;
            const buttonText = this._buildAssemblyItemButtonText(assemblyItem.id);
            const buttonClassnames = this._buildAssemblyItemButtonClassnames(assemblyItem.id);

            return (
                <li
                    key={`assemblyItem-${index}`}
                    className={`listItem listItem-child js-assemblyItem-${assemblyItem.id}`}
                >
                    <div className={'listItem-hd'}>
                        <h3 className={'hdg'}>
                            {item.name}
                        </h3>
                    </div>
                    <div className={'listItem-ft'}>
                        <ul className={'stereo'}>
                            <li>
                                {departmentName}
                            </li>
                            <li>
                                <button
                                    type={'button'}
                                    className={buttonClassnames}
                                    data-department-id={item.department_id}
                                    data-assembly-item-id={assemblyItem.id}
                                    onClick={this.props.onClickAddAssemblyItemHandler}
                                >
                                    {buttonText}
                                </button>
                            </li>
                        </ul>
                    </div>
                </li>
            );
        });

        return assemblyItemListJsx;
    }

    _buildAccordionListClassnames() {
        return classnames({
            vlist: true,
            'vlist-short': true,
            'u-isVisuallyHidden': !this.props.isOpen,
        });
    }

    render() {
        return (
            <li key={`assembly-${this.props.assembly.name}`}>
                {this._buildAssemblyGroupHeaderJsx()}
                <ul className={this._buildAccordionListClassnames()}>
                    {this._buildAssemblyItemList()}
                </ul>
            </li>
        );
    }
}

AssemblyItem.propTypes = {
    assembly: PropTypes.object.isRequired,
    assemblyItemsGroupedByAssemblyId: PropTypes.object.isRequired,
    departmentIdMap: PropTypes.object.isRequired,
    isAssemblyItemSubmitSuccess: PropTypes.bool.isRequired,
    isOpen: PropTypes.bool.isRequired,
    isSubmitAssemblySuccess: PropTypes.bool.isRequired,
    isSubmittingAssembly: PropTypes.bool.isRequired,
    itemIdMap: PropTypes.object.isRequired,
    onClickAccordionTriggerHandler: PropTypes.func.isRequired,
    onClickAddAssemblyHandler: PropTypes.func.isRequired,
    onClickAddAssemblyItemHandler: PropTypes.func.isRequired,
    submittedAssemblyItemId: PropTypes.number.isRequired,
};
