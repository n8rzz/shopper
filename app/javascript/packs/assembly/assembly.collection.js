import AssemblyView from "./assembly.view";

export default class AssemblyCollection {
    get length() {
        return this._items.length;
    }

    constructor(elementList) {
        this._items = [];

        return this._init(elementList);
    }

    _init(elementList) {
        for (let i = 0; i < elementList.length; i++) {
            const element = elementList[i];
            const viewModel = new AssemblyView(element);

            this._add(viewModel);
        }

        return this;
    }

    _add(viewModel) {
        this._items.push(viewModel);
    }
}
