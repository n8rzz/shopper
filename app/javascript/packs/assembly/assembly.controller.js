// import ApiService from '../service/api.service';
import AssemblyCollection from './assembly.collection';

const CLASSNAMES = {
    ROOT_ELEMENT: 'js-assemblyList',
    TRIGGER_ELEMENT: 'js-assemblyList-trigger',
};

export default class AssemblyController {
    constructor() {
        const $element = document.getElementsByClassName(CLASSNAMES.TRIGGER_ELEMENT);
        this._collection = new AssemblyCollection($element);
        this._isEnabled = false;

        if ($element.length === 0) {
            return this.disable();
        }

        return this._init()
            ._createChildren()
            ._setupHandlers()
            .enable();
    }

    _init() {
        return this;
    }

    _createChildren() {
        return this;
    }

    _setupHandlers() {
        return this;
    }

    enable() {
        if (this._isEnabled) {
            return this;
        }

        this._isEnabled = true;

        return this;
    }

    disable() {
        if (!this._isEnabled) {
            return this;
        }

        this._isEnabled = false;

        return this;
    }
}

document.addEventListener('turbolinks:load', () => {
    const assemblyController = new AssemblyController();
});
