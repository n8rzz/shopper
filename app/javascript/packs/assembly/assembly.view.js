const CLASSNAME = {
    ASSEMBLY_ITEM_LIST: 'js-asemblyItemList',
    IS_VISUALLY_HIDDEN: 'u-isVisuallyHidden'
};

export default class AssemblyView {
    constructor($element) {
        const elementId = $element.dataset.assemblyItemId;
        this._$element = $element;
        this._$assemblyItemList = document.getElementsByClassName(`${CLASSNAME.ASSEMBLY_ITEM_LIST}-${elementId}`)[0];

        return this._init()
            ._createChildren()
            ._setupHandlers()
            .enable();
    }

    _init() {
        return this;
    }

    _createChildren() {
        return this;
    }

    _setupHandlers() {
        this._onClickHandler = this._onClick.bind(this);

        return this;
    }

    enable() {
        if (this._isEnabled) {
            return;
        }

        this._$element.addEventListener('click', this._onClickHandler);

        this._isEnabled = true;

        return this;
    }

    disable() {
        if (!this._isEnabled) {
            return;
        }

        this._$element.removeEventListener('click', this._onClickHandler);

        this._isEnabled = false;

        return this;
    }

    _onClick(event) {
        if (!this._hasAssemblyItemList()) {
            return;
        }

        if (!this._isHidden()) {
            this._$assemblyItemList.classList.add(CLASSNAME.IS_VISUALLY_HIDDEN);

            return;
        }

        this._$assemblyItemList.classList.remove(CLASSNAME.IS_VISUALLY_HIDDEN);
    }

    _hasAssemblyItemList() {
        return this._$assemblyItemList != null;
    }

    _isHidden() {
        return this._$assemblyItemList.classList.contains(CLASSNAME.IS_VISUALLY_HIDDEN);
    }
}
