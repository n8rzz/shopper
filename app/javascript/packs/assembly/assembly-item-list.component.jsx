import React from 'react';
import PropTypes from 'prop-types';
import { AssemblyItemContainer } from './assembly-item.container';

export function AssemblyItemList(props) {
    return (
        <ul className={'vlist mix-vlist_isClickable js-accordionList'}>
            {props.assemblies.map((assembly, index) => (
                <AssemblyItemContainer
                    key={`assemblyItemContainer-${index}`}
                    assembly={assembly}
                    assemblyItemsGroupedByAssemblyId={props.assemblyItemsGroupedByAssemblyId}
                    csrf={props.csrf}
                    departmentIdMap={props.departmentIdMap}
                    itemIdMap={props.itemIdMap}
                />
            ))}
        </ul>
    );
}

AssemblyItemList.propTypes = {
    assemblies: PropTypes.array.isRequired,
    assemblyItemsGroupedByAssemblyId: PropTypes.object.isRequired,
    csrf: PropTypes.string.isRequired,
    departmentIdMap: PropTypes.object.isRequired,
    itemIdMap: PropTypes.object.isRequired,
};
