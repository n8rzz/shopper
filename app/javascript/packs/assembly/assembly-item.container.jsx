import React from 'react';
import PropTypes from 'prop-types';
import ApiService from '../service/api.service';
import EventService from '../service/event.service';
import { EVENT_NAME } from '../constants/event-names';
import { AssemblyItem } from './assembly-item.component';

export class AssemblyItemContainer extends React.Component {
    constructor(props) {
        super(props);

        this._onClickAccordionTriggerHandler = this._onClickAccordionTrigger.bind(this);
        this._onClickAddAssemblyHandler = this._onClickAddAssembly.bind(this);
        this._onClickAddAssemblyItemHandler = this._onClickAddAssemblyItem.bind(this);
        this._onSubmitAddAssemblySuccessHandler = this._onSubmitAddAssemblySuccess.bind(this);
        this._onSubmitAddAssemblyItemSuccessHandler = this._onSubmitAddAssemblyItemSuccess.bind(this);

        this.state = {
            isOpen: false,
            isSubmittingAssembly: false,
            isSubmitAssemblySuccess: false,
            submittedAssemblyItemId: -1,
            isAssemblyItemSubmitSuccess: false,
        };
    }

    _onClickAccordionTrigger() {
        this.setState((prevState) => ({ isOpen: !prevState.isOpen }));
    }

    _onClickAddAssembly(event) {
        event.preventDefault();
        event.currentTarget.blur();

        // /order_items/create/assembly
        const assemblyId = this.props.assembly.id;
        // FIXME: move these away from data attributes
        const addAssemblyToOrderUrl = '/order_items/create/assembly.json';
        const requestPayload = {
            assembly_id: assemblyId,
        };

        this.setState({ isSubmittingAssembly: true }, this._onSubmitAddAssembly(addAssemblyToOrderUrl, requestPayload));
    }

    _onSubmitAddAssembly(addAssemblyToOrderUrl, requestPayload) {
        ApiService.post(addAssemblyToOrderUrl, requestPayload, this.props.csrf)
            .then((response) => {
                if (response.status >= 300) {
                    console.error(`
                        Received an unexpected status code from ${addAssemblyToOrderUrl}. See response: ${response}
                    `);

                    return;
                }

                const notice = `${this.props.assembly.name} added to order`;

                EventService.emit(EVENT_NAME.NOTICE_SUCCESS, notice);

                this.setState(
                    { isSubmittingAssembly: false, isSubmitAssemblySuccess: true },
                    this._onSubmitAddAssemblySuccessHandler,
                );
            })
            .catch((error) => { throw error; });
    }

    _onSubmitAddAssemblySuccess() {
        setTimeout(() => {
            this.setState({ isSubmitAssemblySuccess: false });
        }, 3000);
    }

    _onClickAddAssemblyItem(event) {
        event.preventDefault();
        event.currentTarget.blur();

        const assemblyId = this.props.assembly.id;
        // FIXME: move these away from data attributes
        const { assemblyItemId, assemblyItemName, departmentId } = event.currentTarget.dataset;
        const assemblyItemUrl = '/order_items/create/item.json';
        const requestPayload = {
            assembly_id: assemblyId,
            department_id: departmentId,
            item_id: assemblyItemId,
        };

        this.setState(
            { submittedAssemblyItemId: parseInt(assemblyItemId, 10) },
            this._onSubmitAddAssemblyItem(assemblyItemName, assemblyItemUrl, requestPayload),
        );
    }

    _onSubmitAddAssemblyItem(assemblyItemName, assemblyItemUrl, requestPayload) {
        ApiService.post(assemblyItemUrl, requestPayload, this.props.csrf)
            .then((response) => {
                if (response.status >= 300) {
                    console.error(`
                        Received an unexpected status code from ${assemblyItemUrl}. See response: ${response}
                    `);

                    return;
                }

                const notice = `${assemblyItemName} added to order`;

                EventService.emit(EVENT_NAME.NOTICE_SUCCESS, notice);

                this.setState({ isAssemblyItemSubmitSuccess: true }, this._onSubmitAddAssemblyItemSuccessHandler);
            })
            .catch((error) => { throw error; });
    }

    _onSubmitAddAssemblyItemSuccess() {
        setTimeout(() => {
            this.setState({
                isAssemblyItemSubmitSuccess: false,
                submittedAssemblyItemId: -1,
            });
        }, 3000);
    }

    render() {
        return (
            <AssemblyItem
                assembly={this.props.assembly}
                assemblyItemsGroupedByAssemblyId={this.props.assemblyItemsGroupedByAssemblyId}
                departmentIdMap={this.props.departmentIdMap}
                itemIdMap={this.props.itemIdMap}
                isSubmittingAssembly={this.state.isSubmittingAssembly}
                isSubmitAssemblySuccess={this.state.isSubmitAssemblySuccess}
                isAssemblyItemSubmitSuccess={this.state.isAssemblyItemSubmitSuccess}
                submittedAssemblyItemId={this.state.submittedAssemblyItemId}
                isOpen={this.state.isOpen}
                onClickAccordionTriggerHandler={this._onClickAccordionTriggerHandler}
                onClickAddAssemblyHandler={this._onClickAddAssemblyHandler}
                onClickAddAssemblyItemHandler={this._onClickAddAssemblyItemHandler}
            />
        );
    }
}

AssemblyItemContainer.propTypes = {
    assembly: PropTypes.object.isRequired,
    assemblyItemsGroupedByAssemblyId: PropTypes.object.isRequired,
    csrf: PropTypes.string.isRequired,
    departmentIdMap: PropTypes.object.isRequired,
    itemIdMap: PropTypes.object.isRequired,
};
