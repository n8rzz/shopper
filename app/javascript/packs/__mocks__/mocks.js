import { assemblyItemsGroupedByAssemblyIdMock } from './assembly-item.mocks';
import { assemblyMock, assemblyListMock, assemblyMapMock } from './assembly.mock';
import { csrfMock } from './csrf.mock';
import { departmentListMock, departmentMapMock } from './department.mock';
import { itemMock, itemListMock, itemMapMock } from './item.mock';
import { locationDepartmentListMock } from './location-department.mock';
import { locationMock } from './location.mock';
import { orderItemListMock } from './order-item.mock';
import { orderMock, orderLocationNameMock } from './order.mock';

export {
    assemblyItemsGroupedByAssemblyIdMock,
    assemblyMock,
    assemblyListMock,
    assemblyMapMock,
    csrfMock,
    departmentListMock,
    departmentMapMock,
    itemMock,
    itemListMock,
    itemMapMock,
    locationDepartmentListMock,
    locationMock,
    orderItemListMock,
    orderMock,
    orderLocationNameMock,
};
