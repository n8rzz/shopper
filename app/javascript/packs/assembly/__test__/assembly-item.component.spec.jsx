import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';

import { AssemblyItem } from '../assembly-item.component';
import { assemblyMock } from '../../__mocks__/assembly.mock';
import { assemblyItemsGroupedByAssemblyIdMock } from '../../__mocks__/assembly-item.mocks';
import { departmentMapMock } from '../../__mocks__/department.mock';
import { itemMapMock } from '../../__mocks__/item.mock';

describe('AssemblyItem', () => {
    test('renders correctly with valid props', () => {
        const component = shallow(<AssemblyItem
            assembly={assemblyMock}
            assemblyItemsGroupedByAssemblyId={assemblyItemsGroupedByAssemblyIdMock}
            departmentIdMap={departmentMapMock}
            isAssemblyItemSubmitSuccess={false}
            isOpen={false}
            isSubmitAssemblySuccess={false}
            isSubmittingAssembly={false}
            itemIdMap={itemMapMock}
            onClickAccordionTriggerHandler={sinon.spy()}
            onClickAddAssemblyHandler={sinon.spy()}
            onClickAddAssemblyItemHandler={sinon.spy()}
            submittedAssemblyItemId={-1}
        />);

        expect(component).toMatchSnapshot();
    });

    describe('when `isSubmittingAssembly` is true', () => {
        let component = null;
        let assemblyBtn = null;

        beforeEach(() => {
            component = shallow(<AssemblyItem
                assembly={assemblyMock}
                assemblyItemsGroupedByAssemblyId={assemblyItemsGroupedByAssemblyIdMock}
                departmentIdMap={departmentMapMock}
                isAssemblyItemSubmitSuccess={false}
                isOpen={false}
                isSubmitAssemblySuccess={false}
                isSubmittingAssembly={true}
                itemIdMap={itemMapMock}
                onClickAccordionTriggerHandler={sinon.spy()}
                onClickAddAssemblyHandler={sinon.spy()}
                onClickAddAssemblyItemHandler={sinon.spy()}
                submittedAssemblyItemId={-1}
            />);
            assemblyBtn = component.find(`.js-assembly-${assemblyMock.id} button`);
        });

        afterEach(() => {
            component = null;
            assemblyBtn = null;
        });

        test('should render the correct text', () => {
            expect(assemblyBtn.text()).toBe('Adding...');
        });

        test('should render with the correct classnames', () => {
            expect(assemblyBtn.hasClass('mix-btn_isLoading')).toBe(true);
        });
    });

    describe('when `isSubmitAssemblySuccess` is true', () => {
        let component = null;
        let assemblyBtn = null;

        beforeEach(() => {
            component = shallow(<AssemblyItem
                assembly={assemblyMock}
                assemblyItemsGroupedByAssemblyId={assemblyItemsGroupedByAssemblyIdMock}
                departmentIdMap={departmentMapMock}
                isAssemblyItemSubmitSuccess={false}
                isOpen={false}
                isSubmitAssemblySuccess={true}
                isSubmittingAssembly={false}
                itemIdMap={itemMapMock}
                onClickAccordionTriggerHandler={sinon.spy()}
                onClickAddAssemblyHandler={sinon.spy()}
                onClickAddAssemblyItemHandler={sinon.spy()}
                submittedAssemblyItemId={-1}
            />);
            assemblyBtn = component.find(`.js-assembly-${assemblyMock.id} button`);
        });

        afterEach(() => {
            component = null;
            assemblyBtn = null;
        });

        test('should render the correct text', () => {
            expect(assemblyBtn.text()).toBe('Success!');
        });

        test('should render with the correct classnames', () => {
            expect(assemblyBtn.hasClass('mix-btn_isSuccess')).toBe(true);
        });
    });

    describe('when `submittedAssemblyItemId` is not -1', () => {
        let component = null;
        let assemblyItemBtn = null;

        beforeEach(() => {
            component = shallow(<AssemblyItem
                assembly={assemblyMock}
                assemblyItemsGroupedByAssemblyId={assemblyItemsGroupedByAssemblyIdMock}
                departmentIdMap={departmentMapMock}
                isAssemblyItemSubmitSuccess={false}
                isOpen={false}
                isSubmitAssemblySuccess={false}
                isSubmittingAssembly={false}
                itemIdMap={itemMapMock}
                onClickAccordionTriggerHandler={sinon.spy()}
                onClickAddAssemblyHandler={sinon.spy()}
                onClickAddAssemblyItemHandler={sinon.spy()}
                submittedAssemblyItemId={1}
            />);
            assemblyItemBtn = component.find(`.js-assemblyItem-${assemblyMock.id} button`);
        });

        afterEach(() => {
            component = null;
            assemblyItemBtn = null;
        });

        test('should render the correct text', () => {
            expect(assemblyItemBtn.text()).toBe('Adding...');
        });

        test('should render with the correct classnames', () => {
            expect(assemblyItemBtn.hasClass('mix-btn_isLoading')).toBe(true);
        });
    });

    describe('when `isAssemblyItemSubmitSuccess` is true', () => {
        let component = null;
        let assemblyItemBtn = null;

        beforeEach(() => {
            component = shallow(<AssemblyItem
                assembly={assemblyMock}
                assemblyItemsGroupedByAssemblyId={assemblyItemsGroupedByAssemblyIdMock}
                departmentIdMap={departmentMapMock}
                isAssemblyItemSubmitSuccess={true}
                isOpen={false}
                isSubmitAssemblySuccess={false}
                isSubmittingAssembly={false}
                itemIdMap={itemMapMock}
                onClickAccordionTriggerHandler={sinon.spy()}
                onClickAddAssemblyHandler={sinon.spy()}
                onClickAddAssemblyItemHandler={sinon.spy()}
                submittedAssemblyItemId={1}
            />);
            assemblyItemBtn = component.find(`.js-assemblyItem-${assemblyMock.id} button`);
        });

        afterEach(() => {
            component = null;
            assemblyItemBtn = null;
        });

        test('should render the correct text', () => {
            expect(assemblyItemBtn.text()).toBe('Success!');
        });

        test('should render with the correct classnames', () => {
            expect(assemblyItemBtn.hasClass('mix-btn_isSuccess')).toBe(true);
        });
    });
});
