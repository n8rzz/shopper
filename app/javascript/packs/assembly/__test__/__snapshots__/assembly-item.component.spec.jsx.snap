// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssemblyItem renders correctly with valid props 1`] = `
<li
  key="assembly-Souvlaki"
>
  <div
    className="listItem js-assembly-1"
  >
    <div
      className="listItem-hd"
    >
      <h2
        className="hdg hdg_2"
        onClick={[Function]}
      >
        Souvlaki
      </h2>
      <div>
        <button
          className="btn btn-toggle"
          onClick={[Function]}
          type="button"
        >
          Add to order
        </button>
      </div>
    </div>
    <div
      className="listItem-ft"
    >
      <ul
        className="stereo"
      >
        <li>
          Items:
          1
        </li>
        <li>
          <a
            className="link"
            href="/assemblies/1/edit"
          >
            Edit Assembly
          </a>
        </li>
      </ul>
    </div>
  </div>
  <ul
    className="vlist vlist-short u-isVisuallyHidden"
  >
    <li
      className="listItem listItem-child js-assemblyItem-1"
      key="assemblyItem-0"
    >
      <div
        className="listItem-hd"
      >
        <h3
          className="hdg"
        >
          Agave Syrup
        </h3>
      </div>
      <div
        className="listItem-ft"
      >
        <ul
          className="stereo"
        >
          <li>
            Bakery
          </li>
          <li>
            <button
              className="btn btn-toggle"
              data-assembly-item-id={1}
              data-department-id={3}
              data-item-id={13}
              data-item-name="Agave Syrup"
              onClick={[Function]}
              type="button"
            >
              Add to order
            </button>
          </li>
        </ul>
      </div>
    </li>
  </ul>
</li>
`;
